controller/internal/controller/cluster_controller.go

```diff
@@ -42,7 +43,24 @@ type ClusterReconciler struct {
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		WithOptions(controller.Options{MaxConcurrentReconciles: 1}). // Hard requirement for current namespace credentials implementation.
+		// Setting max concurrent reconciles is a hard requirement for current namespace credentials implementation.
+		// Following watch ensures that if namespaced credentials change, all objects that use them will be reconciled.
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
+		Watches(&v1alpha1.NamespaceCredentials{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, nc client.Object) []reconcile.Request {
```

Comment:
Is there no way to DRY up these EnqueueRequestsFromMapFunc handlers?  Maybe generics could help or something?

Reply:
I will give it another try as at the first attempt I had problem with interface for list object that client uses to list resources.

Reply:
It's done now.